### PR TITLE
parquet add write coco

### PR DESCRIPTION
**Pull request by GitEasonXu**
add coco_dataset.py.

### CODE REVIEW

File: pyzoo/zoo/orca/data/image/coco_dataset.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import numpy as np
+from PIL import Image
+from os import path as osp
+from pycocotools.coco import COCO
```
Comment (yangw1234) [inline pyzoo/zoo/orca/data/image/coco_dataset.py:19]: this package have to be installed right?

File: pyzoo/test/zoo/orca/data/test_write_parquet.py [hunk 2]
```diff
@@ -163,6 +163,27 @@ def test_write_voc(orca_context_fixture):
...
+        ann_file = os.path.join(resource_path,
+                                "COCO/annotations/instances_{}.json".format(split_names))
+        output_path = os.path.join(temp_dir, "output_dataset")
+        write_coco(img_path, ann_file, output_path="file://" + output_path)
+        data, schema = ParquetDataset._read_as_dict_rdd("file://" + output_path)
+        data = data.collect()[0]
+        image_path = data["image_id"]
+        with open(image_path, "rb") as f:
+            image_bytes = f.read()
+        assert image_bytes == data['image']
```
Comment (yangw1234) [inline pyzoo/test/zoo/orca/data/test_write_parquet.py:181]: maybe also test on label?

**Comment by yangw1234** [comment]
@GitEasonXu as a general recommendation, I think it is better to rebase on master before developing a new feature and open a PR. This way we can get much more clear commit history.

File: pyzoo/zoo/orca/data/image/coco_dataset.py (hunk 1 above)
Comment (gtfaiwxm) [reply pyzoo/zoo/orca/data/image/coco_dataset.py:19]: yes, the package must be installed.

Comment (gtfaiwxm) [reply pyzoo/zoo/orca/data/image/coco_dataset.py:19]: the package is normal API for reading coco format.

**Comment by GitEasonXu** [comment]
Ok, I'll pay attention to it next time.

File: pyzoo/test/zoo/orca/data/test_write_parquet.py (hunk 2 above)
Comment (gtfaiwxm) [reply pyzoo/test/zoo/orca/data/test_write_parquet.py:181]: The test on image can know if there is an error in reading and writing. I think it is an optional operation.

File: pyzoo/zoo/orca/data/image/coco_dataset.py (hunk 1 above)
Comment (yangw1234) [reply pyzoo/zoo/orca/data/image/coco_dataset.py:19]: Hi @jason-dai , if additional package must be installed do you prefer to add it to `install_required` or add it to `extra_requires`.

image

Comment (yangw1234) [reply pyzoo/zoo/orca/data/image/coco_dataset.py:19]: @GitEasonXu @gtfaiwxm how about we test if users have already installed pycocotools, if not, raise a error to users to install it.

**Comment by yangw1234** [comment]
Could you also add "coco" here https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/zoo/orca/data/image/parquet_dataset.py#L256

**Comment by GitEasonXu** [comment]
OK, we will modify "coco" according your way.

**Pull request closed without merging**